rust/hyperlane-base/src/contract_sync/cursor.rs

```diff
@@ -30,43 +36,40 @@ where
             chunk_size,
             last_tip_update: Instant::now(),
             from: initial_height,
+            eta_calculator: SyncerEtaCalculator::new(initial_height, tip, ETA_TIME_WINDOW),
         })
     }
 
     /// Wait based on how close we are to the tip and update the tip,
     /// i.e. the highest block we may scrape.
     async fn rate_limit(&mut self) -> ChainResult<()> {
+        let update_tip = self.last_tip_update.elapsed() >= Duration::from_secs(30);
         if self.from + self.chunk_size < self.tip {
             // If doing the full chunk wouldn't exceed the already known tip,
             // we don't necessarily need to fetch the new tip. Sleep a tiny bit
```

Comment:
nit: now that we don't return here, this comment should be changed to not mention anything about fetching the new tip